src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -626,7 +623,23 @@ private void WriteMessageFields(BuildMessageEventArgs e, bool writeMessage = tru
             }
         }
 
-        private static BuildEventArgsFieldFlags GetMessageFlags(BuildMessageEventArgs e, BuildEventArgsFieldFlags flags, bool writeMessage = true, bool writeImportance = false)
+        private void WriteArguments(object[] arguments)
+        {
+            if (arguments == null || arguments.Length == 0)
```

Comment:
I'm wondering if it makes more sense to write `0` if arguments.Length == 0, the reason being that if you expect some list of arguments and get nothing, that's unexpected. If you get `0`, that's at least a valid answer.

Reply:
No, we actually can’t do that because the reader doesn’t expect to read the argument count integer if the flags don’t include the Arguments. The Flags enum that we write at the beginning of the record defines whether we’ll be writing the argument block at all, including the argument count. The absense of arguments applies whether it’s null or empty.

Reply:
Also the absense of arguments is the vast majority of cases, so not writing the zero saves a lot of space.